### DISCUSSION TITLE
Import funds for X/P chain to hypervm

### DISCUSSION THREAD

**Discussion by knikos** [ava-labs/hypersdk#683] [Q&A] [answered]
Is there a way to import funds (avax) from x/p chain for example to a hypervm subnet? I was looking at the import-assset action in the tokenvm example but that seems to imply that you can only do that between different tokenvms. 
I am more interested in the equivalent of "cross-chain export/import funds from X to P" for example in the hypervm context.

**Comment by patrick-ogrady** [comment, marked as answer]
There will be a way to send/receive funds from the C-Chain as soon as Avalanche Warp Messaging is supported on the Primary Network (next release). This will allow using AVAX or an ERC-20 from the C-Chain as the fee-paying token on a HyperVM.

It is possible such support could also be added to communicate with the X/P-Chain but that is unlikely to happen in the short-term and most communication will go through the C-Chain.

**Reply by knikos** [reply]
great, thanks for the update

**Comment by kpachhai** [comment]
Since we now have AWM support, is there a documentation on how we could make this happen in tokenvm? We are very interested in the bridging of assets from C-chain to our own subnet developed with hypersdk.
